Formula/kn.rb

```diff
@@ -0,0 +1,37 @@
+class Kn < Formula
+  desc "Command-line interface for managing Knative Serving and Eventing resources"
+  homepage "https://github.com/knative/client"
+  url "https://github.com/knative/client.git",
+      tag:      "v0.24.0",
+      revision: "0bbb3ec9d92665d372d8393ef3036ba7e0c069a4"
+  license "Apache-2.0"
+
+  depends_on "bash" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    ENV["GO111MODULE"] = "on"
+    ENV["TAG"]         = "v#{version}"
+
+    system "./hack/build.sh", "--fast"
+
+    bin.install "kn"
+  end
+
+  test do
+    list_plugins_output = shell_output("#{bin}/kn plugins list")
+    assert_match("No plugins found.", list_plugins_output)
+
+    bash_completion_output = shell_output("#{bin}/kn completion bash")
+    assert_match("# bash completion for kn", bash_completion_output)
+
+    zsh_completion_output = shell_output("#{bin}/kn completion zsh")
+    assert_match("# zsh completion for kn", zsh_completion_output)
+
+    version_output = shell_output("#{bin}/kn version")
+    assert_match("Version:      v#{version}", version_output)
+    assert_match("Build Date:   ", version_output)
+    assert_match("Git Revision: ", version_output)
```

Comment:
could we match on the exact revision as specified at the top of the formula ?

Reply:
That's something I tried to do, but could not find a variable where there revision is stored.

Reply:
The Git revision is part of the output of `kn version` 

```
Version:      v20210721-local-bef27e9e-dirty
Build Date:   2021-07-21 12:38:00
Git Revision: bef27e9e
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.24.0)
* Eventing
  - sources.knative.dev/v1 (knative-eventing v0.24.0)
  - eventing.knative.dev/v1 (knative-eventing v0.24.0)
```

So you could try to match on a regexp like `/^Git Revision: (\w+)$/m` and then compare `$1` with the revision provided initially (this is a Perl regexp, `/m` for multi-line match, not sure about the equivalent in Ruby).